### PR TITLE
Fixed delete command takes one parameter of the name of the channel

### PR DESCRIPTION
**Pull request by dsp9049**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -9,17 +9,19 @@ const events = {
 	MESSAGE_REACTION_ADD: 'messageReactionAdd',
 	MESSAGE_REACTION_REMOVE: 'messageReactionRemove',
 }
+// Constant for which channels to not allow for deletion
+const PERMANENT_CHANNELS = [823936480056705094, 823651605554921502]
```
Comment (mjschuetze102) [inline index.js]: There has got to be a better way to do this... there are so many more channels in prod (and the Ids change between test server and prod)

File: index.js [hunk 2]
```diff
@@ -74,7 +76,8 @@ client.on('message', async msg => {
 				"\n!offline -> Removes your status as the current mentor on duty" +
 				"\n!brb {optional: [minutes until return]} -> Notifies any mentees that use !ping that you are afk and returning soon" +
 				"\n!sos -> In case of emergency, request help from another mentor" +
-				"\n!close -> Removes all existing voice and text channels" +
+				"\n!delete {channel name} -> Deletes specified mentoring channel" +
+			"\n!close -> Removes all existing voice and text channels" +
```
Comment (mjschuetze102) [inline index.js]: Fix formatting

File: index.js [hunk 3]
```diff
@@ -223,6 +235,27 @@ client.on('message', async msg => {
 				channel.delete("closing time *Insert song here*")
 			})
 			createdChannels = []
+		} else if (msg.content.toLowerCase().startsWith("!delete")) {
+			const arg = msg.content.slice(7).trim();
```
Comment (mjschuetze102) [inline index.js]: I would do something like this instead in case of name change
```
cmds = msg.content.split(" ")
```

Comment (mjschuetze102) [reply index.js]: Would also stop !deletet from deleting a channel with t in it

**Review by mjschuetze102 — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (dsp9049) [reply index.js]: Those lines don't do anything and should be deleted

File: index.js [hunk 4]
```diff
@@ -223,6 +235,27 @@ client.on('message', async msg => {
...
+						if (((childChannel.name).toLowerCase()).startsWith(command)) {
+							childChannel.delete("closing time *Insert song here*")
+						}
+					})
+					if ((channel.name).startsWith(command)) {
+						channel.delete("closing time *Insert song here*")
+						msg.channel.send(`${msg.author} deleting ${channel.name}`)
+						createdChannels.splice(i, 1)
+					}
+					i += 1
```
Comment (mjschuetze102) [inline index.js]: A) You are modifying an array while looping through itself, which will probably do bad things
B) Your nested for loop and if statement look very similar and could probably be combined.

File: index.js [hunk 5]
```diff
@@ -265,6 +298,9 @@ client.on('message', async msg => {
 			msg.channel.send("The SSE Mentors are getting a good night rest so they can help you out bright and early tomorrow morning.",
 				{files: ["https://media.tenor.com/images/950960797e3b597b8fedc869eabad846/tenor.gif"]}
 			);
+		} else if (msg.content.toLowerCase().startsWith("!turtle")) {
+			msg.channel.send(":turtle:")
+			msg.delete()
```
Comment (mjschuetze102) [inline index.js]: Make it it's own PR... No sneaking things through on my watch

**Review by mjschuetze102 — CHANGES_REQUESTED** [review]

File: index.js [hunk 6]
```diff
@@ -122,7 +122,6 @@ client.on('message', async msg => {
 			return
 		}
 	}
-
```
Comment (mjschuetze102) [inline index.js]: Add line back in

File: index.js [hunk 7]
```diff
@@ -234,25 +233,28 @@ client.on('message', async msg => {
...
 		} else if (msg.content.toLowerCase().startsWith("!delete")) {
-			const arg = msg.content.slice(7).trim();
-			const command = arg.toLowerCase();
-			if (command.length > 0) {
-				i = 0
-				createdChannels.forEach((channel) => {
-					channel.children.forEach((childChannel) => {
-						if (((childChannel.name).toLowerCase()).startsWith(command)) {
-							childChannel.delete("closing time *Insert song here*")
+			//const arg = msg.content.slice(7).trim();
```
Comment (mjschuetze102) [inline index.js]: Remove comment

File: index.js (hunk 5 above)
Comment (dsp9049) [reply index.js]: darn

File: index.js (hunk 4 above)
Comment (dsp9049) [reply index.js]: a) I will add them to a list and then delete them after
b) they are that way so the category does not get deleted before the channels do: if that happens there is a very high chance of ghosted channels occurring

File: index.js (hunk 6 above)
Comment (dsp9049) [reply index.js]: the tack '-'?

File: index.js (hunk 7 above)
Comment (dsp9049) [reply index.js]: yep, that should be removed

File: index.js (hunk 6 above)
Comment (dsp9049) [reply index.js]: If I add it back in on line 125 it breaks everything. Why is it there and what is its purpose

Comment (dsp9049) [reply index.js]: i am dumb

**Review by mjschuetze102 — DISMISSED** [review]

**Review by mjschuetze102 — CHANGES_REQUESTED** [review]
Fix indentation

**Review by mjschuetze102 — APPROVED** [review]
